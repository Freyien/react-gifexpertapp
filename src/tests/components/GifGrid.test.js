import { shallow } from "enzyme"
import '@testing-library/jest-dom'

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'Naruto';


    test('Debe renderearse de forma correcta', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imágenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/any_image.jpg',
            title: 'Título del gif'
        },{
            id: '123',
            url: 'https://localhost/any_image.jpg',
            title: 'Título del gif'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
    
        const wrapper = shallow( <GifGrid category={ category }/> );

        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);

    })
    
    


})


