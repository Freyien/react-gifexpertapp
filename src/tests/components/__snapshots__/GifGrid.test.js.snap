// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe renderearse de forma correcta 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__slow"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash animate__slow"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
